internal/config/config.go

```diff
@@ -343,6 +343,40 @@ func (c Config) ReadConfig(r io.Reader) (dynOnly bool, err error) {
 	return dynOnly, nil
 }
 
+// RedactSensitiveInfo - removes sensitive information
+// like urls and credentials from the configuration
+func (c Config) RedactSensitiveInfo() {
+	sensitiveConfigKeys := map[string][]string{
+		"notify_amqp":          {"url", "routing_key"},
+		"notify_nsq":           {"nsqd_address"},
+		"notify_postgres":      {"connection_string"},
+		"notify_kafka":         {"sasl_username", "sasl_password", "client_tls_cert", "client_tls_key"},
+		"notify_redis":         {"address", "key", "password"},
+		"notify_mqtt":          {"broker", "password", "username"},
+		"notify_mysql":         {"dsn_string"},
+		"notify_webhook":       {"endpoint", "client_cert", "client_key"},
+		"notify_elasticsearch": {"url", "username", "password"},
+		"notify_nats":          {"address", "username", "password", "token", "cert_authority", "client_cert", "client_key"},
+	}
+
+	for configName, configVals := range c {
+		sensitiveKeys, isSensitiveConfig := sensitiveConfigKeys[configName]
+		if isSensitiveConfig {
+			for name, kvs := range configVals {
+				for _, sensitiveKey := range sensitiveKeys {
+					for i := range kvs {
+						if kvs[i].Key == sensitiveKey && len(kvs[i].Value) > 0 {
+							kvs[i].Value = "*redacted*"
+						}
+					}
+				}
+				configVals[name] = kvs
+			}
+			c[configName] = configVals
```

Comment:
you cannot replace the `(c Config)`  instead `RedactSensitiveInfo()` should return a copy of the config. This is the reason why `Clone()` was implemented. 

```
func (c Config) RedactSensitiveInfo() Config {
     nc := c.Clone()
```

and then modify what you want..

Reply:
Done